op-bindings/bindgen/etherscan_processors.go

```diff
@@ -0,0 +1,260 @@
+		// Assuming proxy deployment was using Arachnid's (or Safe's) deployer,
+		// so first 32 bytes of tx input data is used at salt, and remaining data is the contract init bytecode
+		contract.InitBytecode = string([]rune(proxyDeploymentData)[66:])
+
+		abiFilePath, bytecodeFilePath, err := writeContractArtifacts(tempArtifactsDir, contract.Name, []byte(contract.Abi), []byte(contract.InitBytecode))
+		if err != nil {
+			return err
+		}
+
+		err = genBindingsAndWriteMetadata(&contract, abiFilePath, bytecodeFilePath, goPackageName, metadataOutputDir, contractMetadataFileTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func processUnverifiedDeployments(contracts []etherscanContract, tempArtifactsDir, etherscanApiKey, goPackageName, metadataOutputDir string, apiMaxRetries, apiRetryDelay int, contractMetadataFileTemplate *template.Template) error {
+	for _, contract := range contracts {
+		log.Printf("Generating bindings and metadata for unverified Etherscan contract: %s", contract.Name)
+
+		var err error
+		contract.DeployedBytecode, err = fetchDeployedBytecode(fmt.Sprintf(etherscanGetDeployedBytecodeUrl, contract.DeployedAddress, etherscanApiKey), apiMaxRetries, apiRetryDelay)
+		if err != nil {
+			return err
+		}
+
+		contract.InitBytecode, err = fetchDeploymentData(fmt.Sprintf(etherscanGetTxByHashUrl, contract.DeploymentTxHash, etherscanApiKey), apiMaxRetries, apiRetryDelay)
+		if err != nil {
+			return err
+		}
+
+		abiFilePath, bytecodeFilePath, err := writeContractArtifacts(tempArtifactsDir, contract.Name, []byte(contract.Abi), []byte(contract.InitBytecode))
+		if err != nil {
+			return err
+		}
+
+		err = genBindingsAndWriteMetadata(&contract, abiFilePath, bytecodeFilePath, goPackageName, metadataOutputDir, contractMetadataFileTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func processImmutablesProxyDeployments(contracts []etherscanContract, tempArtifactsDir, etherscanApiKey, goPackageName, metadataOutputDir string, apiMaxRetries, apiRetryDelay int, contractMetadataFileTemplate *template.Template) error {
+	for _, contract := range contracts {
+		log.Printf("Generating bindings and metadata for verified Etherscan contract with immutables and proxy deployment: %s", contract.Name)
+
+		var err error
+		contract.Abi, err = fetchAbi(fmt.Sprintf(etherscanGetAbiUrl, contract.DeployedAddress, etherscanApiKey), apiMaxRetries, apiRetryDelay)
+		if err != nil {
+			return err
+		}
+
+		deploymentTxHash, err := fetchDeploymentTxHash(fmt.Sprintf(etherscanGetDeploymentTxHashUrl, contract.DeployedAddress, etherscanApiKey), apiMaxRetries, apiRetryDelay)
+		if err != nil {
+			return err
+		}
+
+		deploymentWithSalt, err := fetchDeploymentData(fmt.Sprintf(etherscanGetTxByHashUrl, deploymentTxHash, etherscanApiKey), apiMaxRetries, apiRetryDelay)
+		if err != nil {
+			return err
+		}
+		re := regexp.MustCompile(fmt.Sprintf("^0x(%s)", contract.ExpectedSalt))
+		if !re.MatchString(deploymentWithSalt) {
```

Comment:
Is this an etherscan thing? why is the salt expected to be at the beginning of contract init code? Or is this a create2deployer implementation detail?

Reply:
It's specific to create2deployer, I've refactored it to make the salt optional, but if `contract.ExpectedSalt` is set, it still assumes it's at the beginning of the deployment tx data